src/realm/object-store/audit.mm

```diff
@@ -0,0 +1,1381 @@
+////////////////////////////////////////////////////////////////////////////
```

Comment:
This file is obj-c++ because it makes lambdas Just Work with libdispatch rather than having to use the C API. Once the remaining Apple-specific bits are abstracted out it can be changed to a normal c++ file.